src/KM_Utils.pas

```diff
@@ -717,4 +732,93 @@ function GetMultiplicator(aShift: TShiftState): Word;
 end;
 
 
+{
+String functions
+These function are replacements for String functions introduced after XE2 (XE5 probably)
+Names are the same as in new Delphi versions, but with 'Str' prefix
+}
+function StrIndexOf(aStr, aSubStr: String): Integer;
+begin
+  Result := AnsiPos(aSubStr, aStr) - 1;
+end;
+
+
+function StrLastIndexOf(aStr, aSubStr: String): Integer;
+var I: Integer;
+begin
+  Result := -1;
+  for I := 1 to Length(aStr) do
+    if StartsStr(aSubStr, StrSubstring(aStr, I-1)) then
+      Result := I - 1;
+end;
+
+
+function StrStartsWith(aStr, aSubStr: String): Boolean;
+begin
+  Result := StartsStr(aSubStr, aStr);
+end;
+
+
+function StrSubstring(aStr: String; aFrom: Integer): String;
+begin
+  Result := Copy(aStr, aFrom + 1, Length(aStr));
+end;
+
+
+function StrSubstring(aStr: String; aFrom, aLength: Integer): String;
+begin
+  Result := Copy(aStr, aFrom + 1, aLength);
+end;
+
+
+function StrContains(aStr, aSubStr: String): Boolean;
+begin
+  Result := StrIndexOf(aStr, aSubStr) <> -1;
+end;
+
+
+function StrTrimRight(aStr: String; aCharsToTrim: TKMCharArray): String;
+var Found: Boolean;
+    I, J: Integer;
+begin
+  for I := Length(aStr) downto 1 do
+  begin
+    Found := False;
+    for J := Low(aCharsToTrim) to High(aCharsToTrim) do
+    begin
+      if aStr[I] = aCharsToTrim[J] then
+      begin
+        Found := True;
+        Break;
+      end;
+    end;
+    if not Found then
+      Break;
+  end;
+  Result := Copy(aStr, 1, I);
+end;
+
+
+function StrSplitToStrArray(aStr, aDelimiters: String): TStringArray;
+var StrArray: TStringDynArray;
+    I: Integer;
+begin
+  StrArray := SplitString(aStr, aDelimiters);
+  SetLength(Result, Length(StrArray));
+  for I := Low(StrArray) to High(StrArray) do
+    Result[I] := StrArray[I];
+end;
+
+
+function StrSplit(aStr, aDelimiters: String): TStrings;
+var StrArray: TStringDynArray;

```

Comment:
It's bad practice to create object (TStringList) inside and return it as parent class (TStrings). Do we really need it this way? Better to pass TStringList from outside in a parameter.